src/site/content/en/fast/codelab-adapt-video-to-image-serving-based-on-network-quality/index.md

```diff
@@ -0,0 +1,210 @@
+use its `effectiveType` property to decide when to serve a video and when to
+serve an image. `effectiveType` can be `'slow-2g'`, `'2g'`, `'3g'`, or `'4g'`.
+
+## Step 1: Check connection type
+
+The background video is currently specified on line 22 of `index.html` and then
+lazy-loaded in `script.js`.
+
+To load it conditionally, first check if connection type detection is supported.
+In `script.js` add an if statement that tests whether the `navigator.connection`
+object exists and whether it has the `effectiveType` property.
+
+Then add an if statement to check the `effectiveType` of the network. To load
+the video only on the fastest connections, it’s enough to check the connection
+for `4g` and in any other case load the image.
+
+```js
+if (navigator.connection && !!navigator.connection.effectiveType) {
+  if (navigator.connection.effectiveType === '4g') {
+    // video loading code
+  } else {
+    // image loading code
+  }
+}
+```
+
+Wrap the existing video loading code in an else statement, so that video will
+still load in browsers that don’t support the Network Information API.
+
+```js
+if (navigator.connection && !!navigator.connection.effectiveType) {
+  if (navigator.connection.effectiveType === '4g') {
+    // video loading code
+  } else {
+    // image loading code
+  }
+} else {
+  const video = document.getElementById('coverVideo');
+  const videoSource = video.getAttribute('data-src');
+  video.setAttribute('src', videoSource);
+
+  video.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+}
+```
+
+## Step 2: Load video
+
+If the `effectiveType` is `'4g'`, you can use the video loading code from the
+beginning of the codelab.
+
+```js
+if (navigator.connection.effectiveType === '4g') {
+  const video = document.getElementById('coverVideo');
+  const videoSource = video.getAttribute('data-src');
+  video.setAttribute('src', videoSource);
+  video.setAttribute('style', 'height: 100%; width: 100%; display:inline');
+} else {
+  // image loading code
+}
+```
+
+Here’s how the video loading code works: the video source is specified inside
+[data-src attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)
+in `index.html` and then lazy-loaded in `script.js`.
+
+```html
+<video id="coverVideo" autoplay loop muted data-src="https://cdn.glitch.com/b6491350-b058-4eb6-aa6c-55c93122073e%2FMatrix%2C%20Console%2C%20Hacking%2C%20Code.mp4?1551464245607"></video>
+```
+
+Because the video source is specified in the `data-src` attribute, it isn’t
+displayed or downloaded initially.
+[Data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)
+allow you to store extra information on standard HTML elements. They can be
+named anything, as long as it starts with "data-".
+
+To actually display the video on the page, the location from `data-src` needs to
+be set as the `src` attribute of the video element. This is done in `script.js`.
```

Comment:
"the location from ..." maybe rephrase as "you need to copy the value from `data-src` to the video element's `src` attribute."

I think . you can leave off the "This is done in script.js."